tests/arch/arm/arm_interrupt/src/arm_interrupt.c

```diff
@@ -22,13 +98,90 @@ void k_sys_fatal_error_handler(unsigned int reason, const z_arch_esf_t *pEsf)
 
 	if (reason != expected_reason) {
 		printk("Wrong crash type got %d expected %d\n", reason,
-		       expected_reason);
+			expected_reason);
 		k_fatal_halt(reason);
 	}
 
+	if (run_esf_validation) {
+		if (check_esf_matches_expectations(pEsf) == 0) {
+			esf_validation_rv = TC_PASS;
+		}
+		run_esf_validation = 0;
+	}
+
 	expected_reason = -1;
 }
 
+/**
+ * Set ARM registers with a known pattern:
+ *  r0-r12 are set to 0...12, respectively
+ *  r13 (sp) is left untouched
+ *  r14 (pc) will point to the faulting instruction (udf #90)
+ *  r15 (lr) is set to 15 (since a fault takes place, we never use the value)
+ *
+ * Note: Routine was written to be ARMV6M compatible
+ *
+ * In k_sys_fatal_error_handler above we will check that the ESF provided
+ * as a parameter matches these expectations.
+ */
+void set_regs_with_known_pattern(void)
+{
+	__asm__ volatile(
+		"mov r1, #1\n"
+		"mov r2, #2\n"
+		"mov r3, #3\n"
+		"mov r4, #4\n"
+		"mov r5, #5\n"
+		"mov r6, #6\n"
+		"mov r7, #7\n"
+		"mov r0, #8\n"
+		"mov r8, r0\n"
+		"add r0, r0, #1\n"
+		"mov r9, r0\n"
+		"add r0, r0, #1\n"
+		"mov r10, r0\n"
+		"add r0, r0, #1\n"
+		"mov r11, r0\n"
+		"add r0, r0, #1\n"
+		"mov r12, r0\n"
+		"add r0, r0, #3\n"
+		"mov lr, r0\n"
+		"mov r0, #0\n"
+		"udf #90\n"
+	);
+}
+
+void test_arm_esf_collection(void)
```

Comment:
Maybe you could run the test only when CONFIG_EXTRA_EXCEPTION_INFO is defined, otherwise you can skip the test. 
(see how other tests in this suite are "compile-time" skipped)

I suggest this cause I don't see much value on running the test if this Kconfig macro is not defined.

Reply:
hmm, my thought here was this let's us test that the copy of the esf was done correctly for the `CONFIG_EXTRA_EXCEPTION_INFO=n` codepath. Otherwise, there's no check in place that the basic stack frame is populated correctly for that configuration.

Reply:
Alright, no strong opinion here, you can leave that :)